crates/accelerate/src/convert_2q_block_matrix.rs

```diff
@@ -69,17 +43,33 @@ pub fn blocks_to_matrix(
 fn calculate_matrix(
     matrix: ArrayView2<Complex64>,
     q_list: &[usize],
-    swap_gate: &Array2<Complex64>,
     identity: &Array2<Complex64>,
 ) -> Option<Array2<Complex64>> {
     match q_list {
         [0] => Some(kron(identity, &matrix)),
         [1] => Some(kron(&matrix, identity)),
-        [1, 0] => Some(swap_gate.dot(&matrix).dot(swap_gate)),
+        [1, 0] => Some(change_basis(matrix)),
         _ => None,
     }
 }
 
+fn change_basis(matrix: ArrayView2<Complex64>) -> Array2<Complex64> {
+    let mut trans_matrix: Array2<Complex64> = matrix.reversed_axes().to_owned();
+    let mut temp = trans_matrix.slice(s![2_usize, ..]).to_owned();
```

Comment:
I assume this needs to be mutable because you're reusing the variable. But could you just do`let` twice and avoid the mutation here since nothing is writing to temp it feel uncessary to make this mutable and calling `to_owned()` on an `ArrayView` involves extra copies.